reference/format/search.rst

```diff
@@ -1,2 +1,137 @@
 Search Index
 ============
+
++-----------+------------------------------------------------------------+
+| Filename  | ``search.yml``                                             |
++-----------+------------------------------------------------------------+
+| Root Node | The fully qualified class name of the entity being indexed |
++-----------+------------------------------------------------------------+
+| Options   | * `alias`_                                                 |
+|           | * `fields`_                                                |
+|           |                                                            |
+|           |   * :ref:`name <reference-format-search-fields-name>`      |
+|           |   * `relation_fields`_                                     |
+|           |   * `relation_type`_                                       |
+|           |   * `target_fields`_                                       |
+|           |   * `target_type`_                                         |
+|           |                                                            |
+|           | * `label`_                                                 |
+|           | * `mode`_                                                  |
+|           | * `route`_                                                 |
+|           |                                                            |
+|           |   * :ref:`name <reference-format-search-route-name>`       |
+|           |   * `parameters`_                                          |
+|           |                                                            |
+|           | * `search_template`_                                       |
+|           | * `title_fields`_                                          |
++-----------+------------------------------------------------------------+
+
+``alias``
+---------
+
+**type**: ``string``
+
+An alias used in the ``from`` keyword in :ref:`advanced search <advanced-search-api>`.
+
+.. _reference-format-search-fields:
+
+``fields``
+----------
+
+**type**: ``sequence``
+
+The list of fields shown in the search result. Each entry must be a map that can use the following
+keys to configure the behavior of each field:
+
+.. _reference-format-search-fields-name:
+
+``name``
+~~~~~~~~
+
+**type**: ``string``
+
+The name of the entity property. This option is required.
+
+``relation_fields``
+~~~~~~~~~~~~~~~~~~~
+
+**type**: ``sequence``
+
+``relation_type``
+~~~~~~~~~~~~~~~~~
+
+**type**: ``string``
+
+When the property denotes an association with another entity, the type of association (one of
+``one-to-one``, ``many-to-many``, or ``one-to-many``) must be configured with this option.
+
+``target_fields``
+~~~~~~~~~~~~~~~~~
+
+**type**: ``sequence``
+
+When the field represents an association (i.e. a value is configured for `relation_type`_), this is
```

Comment:
Wrong description. `target_fields` is an array of virtual fields for entity field from `name` parameter.
If current field type is string, data from this field additionally will be present in `all_data` virtual field.